Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPISidebar.mm

```diff
@@ -714,6 +719,204 @@
+        @"    .catch(() => browser.test.notifyPass())",
+    ];
+
+    auto *resources = @{
+        @"background.js": Util::constructScript(script),
+        @"sidebar.html": @"<h1>Sidebar</h1>",
+    };
+
+    int presentSidebarCallCount = 0;
+    int *presentSidebarCallCountPtr = &presentSidebarCallCount;
+
+    auto manager = getManagerFor(resources, sidebarActionManifest);
+    manager.get().internalDelegate.presentSidebar = ^(_WKWebExtensionSidebar *) {
+        (*presentSidebarCallCountPtr)++;
+    };
+
+    [manager loadAndRun];
+    EXPECT_EQ(presentSidebarCallCount, 0);
+}
+
+TEST_F(WKWebExtensionAPISidebar, SidebarActionCloseFailsWithoutUserGesture)
+{
+    auto *script = @[
+        @"await browser.sidebarAction.close()",
+        @"    .then(() => browser.test.notifyFail('sidebarAction.close() did not throw without user gesture'))",
+        @"    .catch(() => browser.test.notifyPass())",
+    ];
+
+    auto *resources = @{
+        @"background.js": Util::constructScript(script),
+        @"sidebar.html": @"<h1>Sidebar</h1>",
+    };
+
+    int closeSidebarCallCount = 0;
+    int *closeSidebarCallCountPtr = &closeSidebarCallCount;
+
+    auto manager = getManagerFor(resources, sidebarActionManifest);
+    manager.get().internalDelegate.closeSidebar = ^(_WKWebExtensionSidebar *) {
+        (*closeSidebarCallCountPtr)++;
+    };
+
+    [manager loadAndRun];
+    EXPECT_EQ(closeSidebarCallCount, 0);
+}
+
+TEST_F(WKWebExtensionAPISidebar, SidebarActionToggleFailsWithoutUserGesture)
+{
+    auto *script = @[
+        @"await browser.sidebarAction.toggle()",
+        @"    .then(() => browser.test.notifyFail('sidebarAction.close() did not throw without user gesture'))",
+        @"    .catch(() => browser.test.notifyPass())",
+    ];
+
+    auto *resources = @{
+        @"background.js": Util::constructScript(script),
+        @"sidebar.html": @"<h1>Sidebar</h1>",
+    };
+
+    int openSidebarCallCount = 0;
+    int *openSidebarCallCountPtr = &openSidebarCallCount;
+    int closeSidebarCallCount = 0;
+    int *closeSidebarCallCountPtr = &closeSidebarCallCount;
+
+    auto manager = getManagerFor(resources, sidebarActionManifest);
+    manager.get().internalDelegate.presentSidebar = ^(_WKWebExtensionSidebar *) {
+        (*openSidebarCallCountPtr)++;
+    };
+    manager.get().internalDelegate.closeSidebar = ^(_WKWebExtensionSidebar *) {
+        (*closeSidebarCallCountPtr)++;
+    };
+
+    [manager loadAndRun];
+    EXPECT_EQ(openSidebarCallCount, 0);
+    EXPECT_EQ(closeSidebarCallCount, 0);
+}
+
+TEST_F(WKWebExtensionAPISidebar, SidebarActionOpenSucceedsWithUserGesture)
+{
+    auto *script = @[
+        @"browser.test.yield('Apply user gesture')",
+        @"await browser.sidebarAction.open()",
+        @"    .catch(() => browser.test.notifyFail('sidebarAction.open() threw with user gesture'))",
+        @"    .then(() => browser.test.notifyPass())",
```

Comment:
This needs to use `Util::runScriptWithUserGesture()`. See `TEST(WKWebExtensionAPIPermissions, AcceptPermissionsDenyMatchPatternsRequest)` for an example.